packages/app-studio-toolkit/src/apis/validateLCAP.ts

```diff
@@ -1,19 +1,14 @@
-import { getLogger } from "../logger/logger";
-import { optionalRequire } from "../utils/optional-require";
+import { getLogger } from "@sap/artifact-management";
+import { extensions } from "vscode";
 
-export async function isLCAPEnabled(): Promise<boolean | undefined> {
-  const logger = getLogger().getChildLogger({ label: "isLCAPEnabled" });
-  const noSapPlugin = "NO_SAP_PLUGIN_FOUND";
-  const sapPlugin = optionalRequire<any>("@sap/plugin") ?? noSapPlugin;
+export const LACP_EXTENSION_ID = "SAPSE.lcap-cockpit";
 
-  if (sapPlugin === noSapPlugin) {
-    logger.trace("Failed to load @sap/plugin, so returning undefined.");
-    return;
-  }
-  logger.trace("@sap/plugin successfully loaded.");
+export async function isLCAPEnabled(): Promise<boolean> {
+  const logger = getLogger().getChildLogger({ label: "isLCAPEnabled" });
 
-  const isLCAPEnabled = await sapPlugin.window.isLCAPEnabled();
-  logger.trace("LCAP enabled successfully received.", { isLCAPEnabled });
+  // LCAP mode is determined by the existence of the LCAP extension
+  const isLCAPEnabled = !!extensions.getExtension(LACP_EXTENSION_ID);
+  logger.trace("LCAP enabled mode", { isLCAPEnabled });
 
-  return isLCAPEnabled as boolean | undefined;
+  return Promise.resolve(isLCAPEnabled);
```

Comment:
this function is **async** so an additional promise (`Promise.resolve`) wrapper is not needed.

A non promise value returned from an async promise will be automatically wrapped
in a resolved promise.